Packs/FeedSharedTenantElasticsearch/Integrations/FeedElasticsearch/FeedElasticsearch.py

```diff
@@ -0,0 +1,345 @@
+                if HTTP_ERRORS.get(res.status_code) is not None:
+                    # if it is a known http error - get the message form the preset messages
+                    return_error("Failed to connect. "
+                                 "The following error occurred: {}".format(HTTP_ERRORS.get(res.status_code)))
+
+                else:
+                    # if it is unknown error - get the message from the error itself
+                    return_error("Failed to connect. The following error occurred: {}".format(str(e)))
+
+    except requests.exceptions.RequestException as e:
+        return_error("Failed to connect. Check Server URL field and port number.\nError message: " + str(e))
+
+    demisto.results('ok')
+
+
+def get_indicators_command(client, feed_type, src_val, src_type, default_type):
+    """Implements es-get-indicators command"""
+    now = datetime.now()
+    if feed_type == FEED_TYPE_GENERIC:
+        search = get_scan_generic_format(client, now)
+        get_generic_indicators(search, src_val, src_type, default_type)
+    else:
+        search = get_scan_insight_format(client, now, feed_type=feed_type)
+        get_demisto_indicators(search)
+
+
+def get_generic_indicators(search, src_val, src_type, default_type):
+    """Implements get indicators in generic format"""
+    ioc_lst: list = []
+    for hit in search.scan():
+        hit_lst = extract_indicators_from_generic_hit(hit, src_val, src_type, default_type)
+        ioc_lst.extend(hit_lst)
+    hr = tableToMarkdown('Indicators', ioc_lst, [src_val])
+    return_outputs(hr, {}, ioc_lst)
+
+
+def get_demisto_indicators(search):
+    """Implements get indicators in insight format"""
+    limit = int(demisto.args().get('limit', FETCH_SIZE))
+    indicators_list: list = []
+    ioc_enrch_lst: list = []
+    for hit in search.scan():
+        hit_lst, hit_enrch_lst = extract_indicators_from_insight_hit(hit)
+        indicators_list.extend(hit_lst)
+        ioc_enrch_lst.extend(hit_enrch_lst)
+        if len(indicators_list) >= limit:
+            break
+    hr = tableToMarkdown('Indicators', list(set(map(lambda ioc: ioc.get('name'), indicators_list))), 'Name')
+    if ioc_enrch_lst:
+        for ioc_enrch in ioc_enrch_lst:
+            hr += tableToMarkdown('Enrichment', ioc_enrch, ['value', 'sourceBrand', 'score'])
+    return_outputs(hr, {}, indicators_list)
+
+
+def fetch_indicators_command(client, feed_type, src_val, src_type, default_type, last_fetch):
+    """Implements fetch-indicators command"""
+    last_fetch_timestamp = get_last_fetch_timestamp(last_fetch, client.time_method, client.fetch_time)
+    if feed_type != FEED_TYPE_GENERIC:
+        now_ts = fetch_and_create_indicators_insight_format(client, last_fetch_timestamp)
+    else:
+        now_ts = fetch_and_create_indicators_generic_format(client, src_val, src_type, default_type,
+                                                            last_fetch_timestamp)
```

Comment:
lets do the batching here, or even in the main, you have to many places were you push the data into the system